client-core/src/wallet/default_wallet_client.rs

```diff
@@ -395,6 +395,19 @@ where
         ))
     }
 
+    fn has_unspent_transactions(
+        &self,
+        name: &str,
+        passphrase: &SecUtf8,
+        inputs: &[TxoPointer],
+    ) -> Result<bool> {
+        // Check if wallet exists
+        self.wallet_service.view_key(name, passphrase)?;
```

Comment:
A suggestion: I notice we are using `view_key` to check for wallet existence or passphrase correctness in wallet client, maybe we add a `wallet_service.has_wallet()` ?